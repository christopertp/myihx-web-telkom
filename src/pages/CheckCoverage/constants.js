export const DATA_FETCHED = 'CheckCoverage/data-fetched';
export const FAILED = 'CheckCoverage/failed';
export const LOADING = 'CheckCoverage/loading';
