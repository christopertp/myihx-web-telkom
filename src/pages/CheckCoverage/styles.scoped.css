.map {
  background-color: #ffffff;
  border-radius: 0.5rem;
  box-shadow: 0 0.5rem 1rem rgba(0, 0, 1, 0.15);
  margin: 0 auto;
  padding: 1.5rem;
  width: fit-content;
}

.map > header > h1 {
  font-size: 1.75rem;
  font-weight: 800;
  line-height: 1.2;
  margin: 0.75rem 0 1.5rem;
}

.map > div {
  background-color: #999999;
  height: 37.5rem;
  width: 55rem;
}

.map > section {
  bottom: 2.5rem;
  left: 50%;
  position: absolute;
  transform: translateX(-50%);
  width: 60%;
}

.map > section > input {
  background: url(/assets/ic_search_red.svg);
  background-color: #FFFFFF;
  background-position: 0.75rem;
  background-repeat: no-repeat;
  border: none;
  border-radius: 0.5rem;
  box-shadow: 0 0.25rem 0.625rem 0 rgba(0, 0, 0, 0.2);
  color: var(--light-text-color);
  display: block;
  font-size: 0.9rem;
  font-weight: 400;
  margin-bottom: 1rem;
  outline: none;
  padding: 0.75rem 0.75rem 0.75rem 3rem;
  width: 100%;
}

.map > section > div {
  background-color: #FFFFFF;
  border-radius: 0.5rem;
  box-shadow: 0 0.25rem 0.625rem 0 rgba(0, 0, 0, 0.2);
  margin-bottom: 1rem;
  padding: 0.75rem;
}

.pt-next, .pt-login {
  margin-top: 2rem;
}

.pt-next > *:first-child {
  margin-bottom: 0.5rem;
  width: 100%;
}

.pt-next > *:last-child, .pt-login > *:last-child {
  color: #333435 !important;
  width: 100%;
}

.pt-login {
  display: flex;
  flex-wrap: wrap;
}

.pt-login > a:first-child {
  flex: 1;
  margin-right: 1.25rem;
}

.pt-login > a:nth-child(2) {
  flex: 1
}

.pt-2 {
  text-align: center;
}

.pt-2 > h2 {
  font-size: 1.25rem;
  font-weight: 800;
  line-height: 1.5;
  margin-top: 1.75rem;
}

.pt-2 > p {
  line-height: 1.5;
  margin: 0.5rem 0 2.5rem;
}

.pt-2 > div { display: flex; }
.pt-2 > div > * { flex: 1; }
.pt-2 > div > *:first-child { margin-right: 1rem; }

@media only screen and (max-width: 48em) {
  .map {
    margin: -1rem;
    padding: 0;
  }

  .map > header {
    align-items: center;
    display: flex;
    padding: 1.25rem 1rem;
    width: 100vw;
  }

  .map > header > h1 {
    font-size: 1rem;
    letter-spacing: 0.0625rem;
    margin: 0 auto;
    text-transform: uppercase;
  }

  .map > div {
    height: calc(100vh - 4rem);
    width: 100vw;
  }

  .map > section {
    bottom: 1rem;
    width: calc(100vw - 2rem);
  }

  .pt-next, .pt-login {
    margin: auto 0 1.5rem;
  }

  .pt-login > *:first-child {
    margin-right: 1rem;
  }
}
