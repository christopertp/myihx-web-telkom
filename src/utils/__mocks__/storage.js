export const setToken = jest.fn();
export const getToken = jest.fn(() => true);
export const clearStorages = jest.fn();
export const setExpireTime = jest.fn();
export const checkExpireTime = jest.fn(() => false);
export const setUserData = jest.fn();
export const getUserData = jest.fn(() => ({ profilePicture: 'tes' }));
export const setGuestAddress = jest.fn();
export const getGuestAddress = jest.fn(() => '');
export const clearGuestAddress = jest.fn();
