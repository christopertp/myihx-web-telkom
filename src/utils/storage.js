const EXPIRE_TIME_STORAGE = 'myihx_web_expire_time';
const GUEST_ADDRESS_STORAGE = 'myihx_web_guest_address';
const SESSION_TIME = 'myihx_web_session_time';
const TOKEN_STORAGE = 'myihx_web_access_token';
const USER_DATA_STORAGE = 'myihx_web_user_data';

export function setToken(value) {
  localStorage.setItem(TOKEN_STORAGE, value);
}

export function getToken() {
  return localStorage.getItem(TOKEN_STORAGE);
}

export function clearStorages() {
  localStorage.removeItem(TOKEN_STORAGE);
  localStorage.removeItem(EXPIRE_TIME_STORAGE);
  localStorage.removeItem(USER_DATA_STORAGE);
  localStorage.removeItem(GUEST_ADDRESS_STORAGE);
  localStorage.removeItem(SESSION_TIME);
}

export function setExpireTime(value) {
  localStorage.setItem(EXPIRE_TIME_STORAGE, value * 1000);
}

export function checkExpireTime() {
  const time = new Date().getTime();
  const expire = localStorage.getItem(EXPIRE_TIME_STORAGE) || 0;

  return time > expire;
}

export function setUserData(value) {
  localStorage.setItem(USER_DATA_STORAGE, JSON.stringify(value));
}

export function getUserData() {
  const retval = localStorage.getItem(USER_DATA_STORAGE);

  return JSON.parse(retval) || '';
}

export function setGuestAddress(value) {
  localStorage.setItem(GUEST_ADDRESS_STORAGE, JSON.stringify(value));
}

export function getGuestAddress() {
  const retval = localStorage.getItem(GUEST_ADDRESS_STORAGE);

  return JSON.parse(retval) || '';
}

export function clearGuestAddress() {
  localStorage.removeItem(GUEST_ADDRESS_STORAGE);
}

export function setSessionTime(value) {
  localStorage.setItem(SESSION_TIME, value);
}

export function getSessionTime() {
  const retval = localStorage.getItem(SESSION_TIME);

  return parseInt(retval) || 0;
}
