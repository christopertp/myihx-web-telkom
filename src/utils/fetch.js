import axios from 'axios';
import { getToken } from './storage';

const lng = 'en';
export const BASIC_AUTH = {
  'Authorization': 'Basic dGVsa29tOmRhMWMyNWQ4LTM3YzgtNDFiMS1hZmUyLTQyZGQ0ODI1YmZlYQ==',
  'Accept-Language': lng,
};
export const BEARER_AUTH = {
  'Authorization': `Bearer ${getToken()}`,
  'Accept-Language': lng,
};

export default function fetch(url, method, param1, param2) {
  return new Promise((resolve, reject) => {
    axios[method](url, param1, param2)
      .then(res => resolve(res.data))
      .catch(err => {
        const defaultError = {
          code: 500,
          status: 'error',
          message: 'Failed to fetch data. Please contact developer.'
        };

        if (!err.response) {
          reject(defaultError);
        } else if (!err.response.data) {
          reject(defaultError);
        } else {
          reject(err.response.data);
        }
      });
  });
}

export function getUrl(urls) {
  const mode = process.env.MODE;

  if (mode === 'production') {
    return urls[0];
  }

  if (mode === 'staging') {
    return urls[1];
  }

  return urls[2];
}
