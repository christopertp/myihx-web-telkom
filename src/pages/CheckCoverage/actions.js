import { push } from 'react-router-redux';
import * as order from '../../services/order';
import { availability, guestAvailability } from '../../services/product';
import { addressCities, addressDistricts, addressProvinces, addressStreets } from '../../services/user';
import { updateSyncErrors } from '../../utils/redux-form-actions';
import { clearGuestAddress, getToken, setGuestAddress } from '../../utils/storage';
import { DATA_FETCHED, FAILED, LOADING } from './constants';

export function fetchAddress(key, id1, id2, id3) {
  const apis = {
    cities: addressCities,
    districts: addressDistricts,
    provinces: addressProvinces,
    streets: addressStreets,
  };
  const fields = {
    cities: 'cityCode',
    districts: 'districtCode',
    provinces: 'provinceCode',
    streets: 'streetCode',
  };
  return async dispatch => {
    dispatch(loadingAction(true, key.charAt(0)));
    try {
      const { data } = await apis[key](id1, id2, id3);
      dispatch(dataFetchedAction(data, key));
    } catch (err) {
      const message = err.code === 404 ? 'Data tidak ditemukan' : 'Gagal memuat data';
      dispatch(dataFetchedAction([], key));
      dispatch(updateSyncErrors('checkCoverageAddress', fields[key], message));
    }
  };
}

export function fetchAvailability(payload) {
  return async dispatch => {
    const fn = getToken() ? availability : guestAvailability;
    dispatch(loadingAction(true, 'submit'));
    try {
      const { data } = await fn(payload);
      if (data.status === 'PT1') {
        setGuestAddress({
          ...data.installationAddress,
          addressDescription: payload.installationAddress.addressDescription,
        });
        dispatch(failedAction(''));
        dispatch(push('/check-coverage/pt1'));
      } else {
        dispatch(failedAction('Status unknown'));
      }
    } catch (err) {
      const { data, message } = err;
      if (!data) {
        dispatch(failedAction(message));
      } else if (data.status === 'PT2') {
        dispatch(failedAction('PT2'));
        clearGuestAddress();
      } else if (data.status === 'PT3') {
        dispatch(failedAction(''));
        dispatch(push('/check-coverage/pt3'));
        clearGuestAddress();
      } else {
        dispatch(failedAction(message));
      }
    }
  };
}

export function fetchSubscribe() {
  return async dispatch => {
    dispatch(loadingAction(true, 'submit'));
    try {
      await order.getSubscriptions();
      dispatch(push('/personal-data'));
    } catch (err) {
      dispatch(push('/shop/internet'));
      dispatch(loadingAction(false, 'submit'));
    }
  };
}

export function resetMessage() {
  return failedAction('');
}

function dataFetchedAction(data, key) {
  return { type: DATA_FETCHED, data, key };
}

function failedAction(message) {
  return { type: FAILED, message };
}

function loadingAction(isLoading, key) {
  return { type: LOADING, isLoading, key };
}
