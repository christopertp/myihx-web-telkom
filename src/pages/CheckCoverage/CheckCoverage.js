import React, { createContext, useContext, useEffect, useState } from 'react';
import { useDispatch, useSelector } from 'react-redux';
import { useHistory, useParams } from 'react-router-dom';
import Button from '../../components/elements/Button';
import IconButton from '../../components/elements/IconButton';
import Modal from '../../components/elements/Modal';
import ModalLogin from '../../components/elements/ModalLogin';
import AddressForm from '../../components/forms/CheckCoverageAddress';
import WhiteCardPlain from '../../components/layouts/WhiteCardPlain';
import WhiteCardSuccessPlain from '../../components/layouts/WhiteCardSuccessPlain';
import useGMap from '../../hooks/useGMap';
import { getToken } from '../../utils/storage';
import { fetchAvailability, resetMessage, fetchSubscribe } from './actions';
import styles from './styles.scoped.css';

const Context = createContext({});
const DEFAULT_COORD = { lat: -6.1801091, lng: 106.8263765 };

export default function CheckCoverage() {
  const dispatch = useDispatch();
  const [modalLogin, setModalLogin] = useState(false);
  const [address, setAddress] = useState('');
  const [coord, setCoord] = useState(DEFAULT_COORD);
  const history = useHistory();
  const { page } = useParams();

  useEffect(() => {
    page && history.replace('/check-coverage');
    return () => dispatch(resetMessage());
  }, []);

  const contextValue = {
    address, setAddress,
    coord, setCoord,
    setModalLogin,
  };

  return (
    <Context.Provider value={contextValue}>
      {(p => {
        switch (p) {
          case 'address': return <Address />;
          case 'pt1': return <PT1 />;
          case 'pt3': return <PT3 />;
          default: return <MapSelect />;
        }
      })(page)}
      <ModalPT2 />
      <ModalLogin onClose={() => setModalLogin(false)} open={modalLogin} />
    </Context.Provider>
  );
}

export function MapSelect() {
  const { address, setAddress, coord, setCoord } = useContext(Context);
  const mapProps = { coord, setCoord, setAddress };
  const { refInput, refMap } = useGMap(mapProps);

  const setOverFlow = v => {
    const app = document.getElementById('app');
    app.style.overflowY = v;
  };

  useEffect(() => () => setOverFlow(''), []);

  return (
    <section className={styles.map}>
      <header>
        <IconButton name="back_red" to="/shop/internet" />
        <h1>Place Pin Point</h1>
      </header>
      <div ref={refMap} />
      <section>
        <input onBlur={() => setOverFlow('')} onFocus={() => setOverFlow('hidden')} ref={refInput} />
        {address && <div>{address}</div>}
        <Button to="/check-coverage/address">Confirm</Button>
      </section>
    </section>
  );
}

export function Address() {
  const { address, coord } = useContext(Context);
  const dispatch = useDispatch();
  const subtitle = 'Make sure you have entered the right address and location pin point.';

  const onSubmit = installationAddress => {
    const payload = {
      installationAddress,
      latitude: `${coord.lat}`,
      longitude: `${coord.lng}`,
    };
    dispatch(fetchAvailability(payload));
  };

  return (
    <WhiteCardPlain back="/check-coverage" subtitle={subtitle} title="Enter Installation Address">
      <AddressForm address={address} onSubmit={onSubmit} />
    </WhiteCardPlain>
  );
}

export function PT1() {
  const { setModalLogin } = useContext(Context);
  const dispatch = useDispatch();
  const isLoggedIn = !!getToken();
  const title = 'IndiHome service is available in your area!';
  const subtitle = 'Congratulation! Your location is within IndiHome area coverage. Browse packages and activate it in a few easy steps!';
  const onClick = () => setModalLogin(true);
  const onClickSubscribe = () => dispatch(fetchSubscribe());
  const nextButtonProps = isLoggedIn ? { onClick: onClickSubscribe } : { onClick };
  return (
    <WhiteCardSuccessPlain icon="check" subtitle={subtitle} title={title}>
      <div className={styles['pt-next']}>
        <Button {...nextButtonProps}>Next</Button>
        <Button to="/shop/internet" variant="text">Find the best package for me</Button>
      </div>
    </WhiteCardSuccessPlain>
  );
}

export function PT3() {
  const isLoggedIn = !!getToken();
  const title = 'Sorry, IndiHome is not yet available in your area';
  const subtitle = isLoggedIn ? 'We will contact you when itâ€™s available!' : 'Login or register to stay updated!';
  return (
    <WhiteCardSuccessPlain icon="status_failed" subtitle={subtitle} title={title}>
      {isLoggedIn ?
        (
          <div className={styles['pt-next']}>
            <Button to="/check-coverage">Check Other address</Button>
            <Button to="/" variant="text">Back to Home</Button>
          </div>
        ) :
        (
          <div className={styles['pt-login']}>
            <Button to="/login" variant="bordered">Login</Button>
            <Button to="/register">Register</Button>
            <Button to="/" variant="text">Back to Home</Button>
          </div>
        )
      }
    </WhiteCardSuccessPlain>
  );
}

export function ModalPT2() {
  const { setModalLogin } = useContext(Context);
  const dispatch = useDispatch();
  const { message, resetMessage } = useSelector(s => s.checkCoverage);
  const isLoggedIn = !!getToken();
  const onClick = () => setModalLogin(true);
  const nextButtonProps = isLoggedIn ? { to: '/shop/internet' } : { onClick };
  const onCloseModal = () => {
    dispatch(resetMessage());
  };
  return (
    <Modal className={styles['pt-2']} onClose={onCloseModal} open={message === 'PT2'}>
      <h2>Location is a bit far...</h2>
      <p>
        Your address is far from IndiHome area coverage,
        you may have to wait up to 14 days for installation
      </p>
      <div>
        <Button onClick={onCloseModal} variant="bordered">Cancel</Button>
        <Button {...nextButtonProps}>Continue</Button>
      </div>
    </Modal>
  );
}
