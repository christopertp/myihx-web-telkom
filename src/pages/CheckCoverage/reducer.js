import { fetchAddress, resetMessage } from './actions';
import { DATA_FETCHED, FAILED, LOADING } from './constants';

const initialState = {
  fetchAddress,
  isLoading: {
    p: true,
    c: false,
    d: false,
    s: false,
    submit: false,
  },
  message: '',
  provinces: [],
  cities: [],
  districts: [],
  resetMessage,
  streets: [],
};

export default function reducer(state = initialState, action = {}) {
  const { type, data, isLoading, key, message } = action;

  switch (type) {
    case DATA_FETCHED:
      return {
        ...state,
        isLoading: { ...state.isLoading, [key.charAt(0)]: false },
        [key]: data,
      };
    case FAILED:
      return {
        ...state,
        isLoading: { ...state.isLoading, submit: false },
        message,
      };
    case LOADING:
      return {
        ...state,
        isLoading: { ...state.isLoading, [key]: isLoading },
      };
    default:
      return state;
  }
}
