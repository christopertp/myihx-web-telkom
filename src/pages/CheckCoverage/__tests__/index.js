import React, { useContext, useEffect, useState } from 'react';
import { useDispatch, useSelector } from 'react-redux';
import { useHistory, useParams } from 'react-router-dom';
import ShallowRenderer from 'react-test-renderer/shallow';
import { getToken } from '../../../utils/storage';
import CheckCoverage, { MapSelect, Address, PT1, PT3, ModalPT2 } from '../CheckCoverage';

jest.mock('../actions', () => ({
  fetchAvailability: jest.fn(),
  fetchSubscribe: jest.fn(),
  resetMessage: jest.fn(),
}));

jest.mock('../../../hooks/useGMap', () => () => ({
  refInput: 'input',
  refMap: 'map',
}));

jest.mock('../../../utils/storage');

useSelector.mockImplementation(fn => {
  fn({});
  return {};
});

const app = {
  style: { overflowY: 'overflow' },
};
document.getElementById = () => (app);

describe('src/pages/CheckCoverage', () => {
  test('render', () => {
    const shallow = new ShallowRenderer();
    const tree = shallow.render(<CheckCoverage />);
    expect(tree).toMatchSnapshot();
  });

  test('render subpage', () => {
    const replace = jest.fn();
    const setModalLogin = jest.fn();
    useHistory.mockImplementationOnce(() => ({ replace }));
    useParams.mockImplementationOnce(() => ({ page: 'address' }));
    useState.mockImplementationOnce(v => [v, setModalLogin]);
    const result1 = CheckCoverage();
    result1.props.children[2].props.onClose();
    expect(result1.props.children[0].type).toBe(Address);
    expect(replace).toHaveBeenCalledWith('/check-coverage');
    expect(setModalLogin).toHaveBeenCalledWith(false);

    useParams.mockImplementationOnce(() => ({ page: 'pt1' }));
    const result2 = CheckCoverage();
    expect(result2.props.children[0].type).toBe(PT1);

    useParams.mockImplementationOnce(() => ({ page: 'pt3' }));
    useEffect.mockImplementationOnce(f => f()());

    const dispatch = jest.fn();
    useDispatch.mockImplementationOnce(() => dispatch);

    const result3 = CheckCoverage();
    expect(result3.props.children[0].type).toBe(PT3);
    expect(dispatch).toHaveBeenCalled();
  });

  test('MapSelect', () => {
    const shallow = new ShallowRenderer();
    const tree = shallow.render(<MapSelect />);
    expect(tree).toMatchSnapshot();
  });

  test('MapSelect input events', () => {
    useContext.mockImplementationOnce(() => ({ address: 'test' }));
    useEffect.mockImplementationOnce(fn => fn()());
    const result = MapSelect();
    const input = result.props.children[2].props.children[0];
    expect(app.style.overflowY).toBe('');
    input.props.onFocus();
    expect(app.style.overflowY).toBe('hidden');
    input.props.onBlur();
    expect(app.style.overflowY).toBe('');
    expect(result.props.children[2].props.children[1].props.children).toBe('test');
  });

  test('Address', () => {
    const shallow = new ShallowRenderer();
    const tree = shallow.render(<Address />);
    expect(tree).toMatchSnapshot();
  });

  test('Address onSubmit', () => {
    const coord = { lat: 1, lng: 2 };
    const dispatch = jest.fn();
    useContext.mockImplementationOnce(() => ({ coord }));
    useDispatch.mockImplementationOnce(() => dispatch);
    const result = Address();
    result.props.children.props.onSubmit('payload');
    expect(dispatch).toHaveBeenCalled();
  });

  test('PT1', () => {
    const shallow = new ShallowRenderer();
    const tree = shallow.render(<PT1 />);
    expect(tree).toMatchSnapshot();
  });

  test('PT1 onClick as guest, onClickSubscribe', () => {
    const setModalLogin = jest.fn();
    useContext.mockImplementationOnce(() => ({ setModalLogin }));
    getToken.mockImplementationOnce(() => false);
    const result = PT1();
    result.props.children.props.children[0].props.onClick();
    expect(setModalLogin).toHaveBeenCalledWith(true);

    const dispatch = jest.fn();
    useDispatch.mockImplementationOnce(() => dispatch);

    const resul2 = PT1();
    resul2.props.children.props.children[0].props.onClick();
    expect(dispatch).toHaveBeenCalled();

  });

  test('PT3', () => {
    const shallow = new ShallowRenderer();
    const tree = shallow.render(<PT3 />);
    expect(tree).toMatchSnapshot();
  });

  test('PT3 as guest', () => {
    getToken.mockImplementationOnce(() => false);
    const result = PT3();
    expect(result.props.subtitle).toBe('Login or register to stay updated!');
    expect(result.props.children.props.children.length).toBe(3);
  });

  test('ModalPT2', () => {
    const shallow = new ShallowRenderer();
    const tree = shallow.render(<ModalPT2 />);
    expect(tree).toMatchSnapshot();
  });

  test('ModalPT2 events & as guest', () => {
    const setModalLogin = jest.fn();
    const dispatch = jest.fn();
    const resetMessage = jest.fn();
    useContext.mockImplementationOnce(() => ({ setModalLogin }));
    useDispatch.mockImplementationOnce(() => dispatch);
    useSelector.mockImplementationOnce(() => ({ resetMessage }));
    getToken.mockImplementationOnce(() => false);
    const result = ModalPT2();
    result.props.onClose();
    expect(resetMessage).toHaveBeenCalled();
    expect(dispatch).toHaveBeenCalled();
    result.props.children[2].props.children[1].props.onClick();
    expect(setModalLogin).toHaveBeenCalledWith(true);
  });
});
