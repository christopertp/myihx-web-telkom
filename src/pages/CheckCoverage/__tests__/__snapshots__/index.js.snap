// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`src/pages/CheckCoverage Address 1`] = `
<WhiteCardPlain
  back="/check-coverage"
  close=""
  help={false}
  icon=""
  size="large"
  subtitle="Make sure you have entered the right address and location pin point."
  title="Enter Installation Address"
>
  <ReduxForm
    destroyOnUnmount={true}
    enableReinitialize={false}
    forceUnregisterOnUnmount={false}
    form="checkCoverageAddress"
    getFormState={[Function]}
    initialValues={
      Object {
        "addressDescription": "",
        "cityCode": "",
        "districtCode": "",
        "postalCode": "",
        "provinceCode": "",
        "rtRw": "",
        "streetCode": "",
      }
    }
    keepDirtyOnReinitialize={false}
    onSubmit={[Function]}
    persistentSubmitErrors={false}
    pure={true}
    shouldAsyncValidate={[Function]}
    shouldError={[Function]}
    shouldValidate={[Function]}
    shouldWarn={[Function]}
    submitAsSideEffect={false}
    touchOnBlur={true}
    touchOnChange={false}
    updateUnregisteredFields={false}
    validate={[Function]}
  />
</WhiteCardPlain>
`;

exports[`src/pages/CheckCoverage MapSelect 1`] = `
<section>
  <header>
    <IconButton
      className=""
      name="back_red"
      onClick={[Function]}
      title=""
      to="/shop/internet"
      type="button"
    />
    <h1>
      Place Pin Point
    </h1>
  </header>
  <div />
  <section>
    <input
      onBlur={[Function]}
      onFocus={[Function]}
    />
    <Button
      className=""
      disabled={false}
      onClick={null}
      to="/check-coverage/address"
      type="button"
      variant="red"
    >
      Confirm
    </Button>
  </section>
</section>
`;

exports[`src/pages/CheckCoverage ModalPT2 1`] = `
<Modal
  className=""
  onClose={[Function]}
  open={false}
>
  <h2>
    Location is a bit far...
  </h2>
  <p>
    Your address is far from IndiHome area coverage, you may have to wait up to 14 days for installation
  </p>
  <div>
    <Button
      className=""
      disabled={false}
      onClick={[Function]}
      to=""
      type="button"
      variant="bordered"
    >
      Cancel
    </Button>
    <Button
      className=""
      disabled={false}
      onClick={null}
      to="/shop/internet"
      type="button"
      variant="red"
    >
      Continue
    </Button>
  </div>
</Modal>
`;

exports[`src/pages/CheckCoverage PT1 1`] = `
<WhiteCardSuccessPlain
  close=""
  icon="check"
  subtitle="Congratulation! Your location is within IndiHome area coverage. Browse packages and activate it in a few easy steps!"
  title="IndiHome service is available in your area!"
>
  <div>
    <Button
      className=""
      disabled={false}
      onClick={[Function]}
      to=""
      type="button"
      variant="red"
    >
      Next
    </Button>
    <Button
      className=""
      disabled={false}
      onClick={null}
      to="/shop/internet"
      type="button"
      variant="text"
    >
      Find the best package for me
    </Button>
  </div>
</WhiteCardSuccessPlain>
`;

exports[`src/pages/CheckCoverage PT3 1`] = `
<WhiteCardSuccessPlain
  close=""
  icon="status_failed"
  subtitle="We will contact you when itâ€™s available!"
  title="Sorry, IndiHome is not yet available in your area"
>
  <div>
    <Button
      className=""
      disabled={false}
      onClick={null}
      to="/check-coverage"
      type="button"
      variant="red"
    >
      Check Other address
    </Button>
    <Button
      className=""
      disabled={false}
      onClick={null}
      to="/"
      type="button"
      variant="text"
    >
      Back to Home
    </Button>
  </div>
</WhiteCardSuccessPlain>
`;

exports[`src/pages/CheckCoverage render 1`] = `
<Provider
  value={
    Object {
      "address": "",
      "coord": Object {
        "lat": -6.1801091,
        "lng": 106.8263765,
      },
      "setAddress": [MockFunction],
      "setCoord": [MockFunction],
      "setModalLogin": [MockFunction],
    }
  }
>
  <MapSelect />
  <ModalPT2 />
  <ModalLogin
    onClose={[Function]}
    open={false}
  />
</Provider>
`;
